app/modules/sightings/models.py

```diff
@@ -314,28 +300,64 @@ def any_jobs_active(self):
                 return True
         return False
 
+    @classmethod
+    def get_all_jobs_debug(cls, verbose):
+        jobs = []
+
+        for sighting in Sighting.query.all():
+            jobs.extend(sighting.get_job_debug(annotation_id=None, verbose=verbose))
+        return jobs
+
     # Build up dict to print out status (calling function chooses what to collect and print)
-    def get_job_details(self, annotation_id, verbose):
+    def get_job_debug(self, annotation_id=None, verbose=True):
 
-        details = {}
+        details = []
         for job_id in self.jobs.keys():
-            if annotation_id == self.jobs[job_id]['annotation']:
-                details[job_id] = self.jobs[job_id]
-                if verbose:
-                    details[job_id]['request'] = self.build_identification_request(
-                        self.jobs[job_id]['matching_set'],
-                        self.jobs[job_id]['annotation'],
-                        job_id,
-                        self.jobs[job_id]['algorithm'],
-                    )
-                    details[job_id][
-                        'response'
-                    ] = current_app.acm.request_passthrough_result(
+            if annotation_id and str(annotation_id) != self.jobs[job_id]['annotation']:
+                continue
+            details.append(self.jobs[job_id])
+            details[-1]['type'] = 'Sighting'
+            details[-1]['object_guid'] = self.guid
+            details[-1]['job_id'] = job_id
+
+            if verbose:
+                details[-1]['request'] = self.build_identification_request(
+                    self.jobs[job_id]['matching_set'],
+                    self.jobs[job_id]['annotation'],
+                    job_id,
+                    self.jobs[job_id]['algorithm'],
+                )
+                try:
+                    acm_data = current_app.acm.request_passthrough_result(
                         'job.response', 'post', {}, job_id
                     )
+                    # cm_dict is enormous and as we don't use it in Houston, don't print it as debug
+                    if 'json_result' in acm_data and isinstance(
+                        acm_data['json_result'], dict
+                    ):
+                        acm_data['json_result'].pop('cm_dict', None)
+                    details[-1]['response'] = acm_data
+
+                except HoustonException as ex:
+                    # acm seems particularly flaky for getting the sighting data, if it fails, don't pass it back
+                    details[-1][
+                        'response'
+                    ] = f'Failed to read data from Sage {ex.message}'
 
         return details
 
+    def get_jobs_json(self):
+        job_data = []
+        for job in self.jobs:
```

Comment:
i think this will need a check for `None` for `self.jobs` (e.g. migrated data)

Reply:
But wouldn't that mean that the for loop would match to finding nothing. 
Oh and that's not just migrated data, it's all sightings right now that do not have id config data when they're created. 
I guess I shoudl test that scenario